input/kafkamdm/kafkamdm.go

```diff
@@ -251,6 +254,10 @@ func (k *KafkaMdm) consumePartition(topic string, partition int32, currentOffset
 			if err := offsetMgr.Commit(topic, partition, currentOffset); err != nil {
 				log.Error(3, "kafka-mdm failed to commit offset for %s:%d, %s", topic, partition, err)
 			}
+			if currentOffset < 0 {
+				// we have not yet consumed any messages.
+				continue
```

Comment:
ok, i understand what you are trying to say.  If we never start consuming, we will never lookup offset.Newest.
That is true and as you mentioned unlikely.  But it wouldnt hurt to always capture the logSize.